diff --git a/nvidia/linux_nvswitch.c b/nvidia/linux_nvswitch.c
index bee0bbc..f7e3306 100644
--- a/nvidia/linux_nvswitch.c
+++ b/nvidia/linux_nvswitch.c
@@ -2038,7 +2038,7 @@ nvswitch_os_read_registry_dword
         return -NVL_ERR_GENERIC;
     }
 
-    strncpy(regkey_val, regkey_val_start, regkey_val_len);
+    memcpy(regkey_val, regkey_val_start, regkey_val_len);
     regkey_val[regkey_val_len] = '\0';
 
     if (nvswitch_os_strtouint(regkey_val, data) != 0)
@@ -2393,7 +2393,7 @@ nvswitch_os_strncpy
     NvLength length
 )
 {
-    return strncpy(dest, src, length);
+    return memcpy(dest, src, length);
 }
 
 int
diff --git a/nvidia/os-interface.c b/nvidia/os-interface.c
index 62279a2..6b674a2 100644
--- a/nvidia/os-interface.c
+++ b/nvidia/os-interface.c
@@ -729,7 +729,7 @@ NvU32 NV_API_CALL os_get_current_process(void)
 void NV_API_CALL os_get_current_process_name(char *buf, NvU32 len)
 {
     task_lock(current);
-    strncpy(buf, current->comm, len - 1);
+    strscpy(buf, current->comm, len - 1);
     buf[len - 1] = '\0';
     task_unlock(current);
 }
diff --git a/nvidia-modeset/nvidia-modeset-linux.c b/nvidia-modeset/nvidia-modeset-linux.c
index 33e9167..fc2162a 100644
--- a/nvidia-modeset/nvidia-modeset-linux.c
+++ b/nvidia-modeset/nvidia-modeset-linux.c
@@ -666,7 +666,7 @@ int nvkms_strcmp(const char *s1, const char *s2)
 
 char* nvkms_strncpy(char *dest, const char *src, size_t n)
 {
-    return strncpy(dest, src, n);
+    return memcpy(dest, src, n);
 }
 
 void nvkms_usleep(NvU64 usec)
diff --git a/nvidia-uvm/uvm_pmm_gpu.c b/nvidia-uvm/uvm_pmm_gpu.c
index 34e56f2..1fda522 100644
--- a/nvidia-uvm/uvm_pmm_gpu.c
+++ b/nvidia-uvm/uvm_pmm_gpu.c
@@ -2804,7 +2804,7 @@ static NV_STATUS init_chunk_split_cache_level(uvm_pmm_gpu_t *pmm, size_t level)
             size_t size;
             size_t align;
             if (level == 0) {
-                strncpy(chunk_split_cache[level].name, "uvm_gpu_chunk_t", sizeof(chunk_split_cache[level].name) - 1);
+                strscpy(chunk_split_cache[level].name, "uvm_gpu_chunk_t", sizeof(chunk_split_cache[level].name) - 1);
                 size = sizeof(uvm_gpu_chunk_t);
                 align = __alignof__(uvm_gpu_chunk_t);
             } else {
